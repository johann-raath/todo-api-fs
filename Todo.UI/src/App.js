import './App.scss';
import { BrowserRouter as Router, Route, useHistory } from "react-router-dom";
import { useState, useMemo, useEffect } from "react";
import { UserContext } from "./context/userContext";
import { ListContext } from './context/listContext';
import { ErrorContext } from './context/errorContext';
import { Container } from "@material-ui/core";
import { Header } from "./components/header/header";
import { Login } from "./pages/login/login";
import { List } from "./pages/todos/list/list";
import { Create } from "./pages/todos/create/create";
import { Edit } from "./pages/todos/edit/edit";
import { Register } from './pages/register/register';
import { getExpiry, getUser, removeUserSession } from "./utils/common";
import { Errors } from './components/errors/errors';
import { LoaderContext } from './context/loaderContext';
import { Loader } from './components/loader/loader';

function App() {

  const history = useHistory();

  const [user, setUser] = useState(null);
  const userValue = useMemo(() => ({ user, setUser }), [user, setUser]);

  const [errors, setErrors] = useState(null);
  const errorsValue = useMemo(() => ({ errors, setErrors }), [errors, setErrors]);

  const [loading, setLoading] = useState(false);
  const loaderValue = useMemo(() => ({ loading, setLoading }), [loading, setLoading]);

  useEffect(() => {
    var dateNow = new Date();
    var expiry = new Date(getExpiry());

    if (expiry > dateNow) {
      setUser(getUser());
    } else {
      removeUserSession();
      setUser(null);
      history.push('/');
    }
  }, []);

  return (
    <Container id="todo">
      <Router>
        <UserContext.Provider value={userValue}>
          <Header></Header>
          <LoaderContext.Provider value={loaderValue}>
            {loaderValue.loading ? (
              <Loader />
            ) : (
                <ErrorContext.Provider value={errorsValue}>
                  <Errors />
                  {userValue.user ? (
                    <div>
                      <Route path="/" exact component={List}></Route>
                      <Route path="/todo/create" component={Create}></Route>
                      <Route path="/todo/edit/:id" component={Edit}></Route>
                    </div>
                  ) : (
                      <div>
                        <Route path="/" exact component={Login}></Route>
                        <Route path="/login" component={Login}></Route>
                        <Route path="/register" component={Register}></Route>
                      </div>
                    )}
                </ErrorContext.Provider>
              )
            }
          </LoaderContext.Provider>
        </UserContext.Provider>
      </Router>
    </Container>
  );
}

export default App;
