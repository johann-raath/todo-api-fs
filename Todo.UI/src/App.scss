#todo {
    .MuiButton-root .MuiButton-label a {
        color: #fff;
        text-decoration: none;
    }

    .MuiFormGroup-root {
        & > * + * {
            margin-top: 30px;
        }
    }
}
