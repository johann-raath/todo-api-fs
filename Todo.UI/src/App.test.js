import ReactDOM from 'react-dom'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";
import App from './App';

Enzyme.configure({ adapter: new Adapter() })

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
})

it('shallow renders app without crashing', () => {
  shallow(<App />);
});

it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
