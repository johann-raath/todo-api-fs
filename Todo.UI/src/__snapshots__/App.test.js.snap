// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
  id="todo"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
  >
    <h1>
      TODO
    </h1>
    <div
      className="userActions"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </span>
      </button>
    </div>
  </div>
  <div />
  <div>
    <div
      className="login"
    >
      <h2>
        Login
      </h2>
      <div
        className="MuiFormGroup-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="userName"
            id="userName-label"
          >
            User Name
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="userName"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  User Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="userPassword"
            id="userPassword-label"
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="current-password"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="userPassword"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Login
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
